server/neptune/workflows/internal/deploy/terraform/state.go

```diff
@@ -39,6 +48,35 @@ func (n *StateReceiver) Receive(ctx workflow.Context, c workflow.ReceiveChannel,
 		}
 	}
 
+	if workflowState.Apply != nil &&
+		workflowState.Apply.Status == state.WaitingJobStatus &&
+		!reflect.ValueOf(workflowState.Apply.OnWaitingActions).IsZero() {
+		// update queue with information about current deployment pending confirm/reject action
+		infos := n.Queue.GetOrderedMergedItems()
+
+		revisionsSummary := n.Queue.GetQueuedRevisionsSummary()
+		state := github.CheckRunQueued
+		runLink := github.BuildRunURLMarkdown(deploymentInfo.Repo.GetFullName(), deploymentInfo.Commit.Revision, deploymentInfo.CheckRunID)
```

Comment:
what is `deploymentInfo` in this context?  is it essentially the item at the top of the queue?

Reply:
deploymentInfo contains all the metadata about the deploy job (this is what gets stored in the queue).
```
type DeploymentInfo struct {
	ID             uuid.UUID
	CheckRunID     int64
	Commit         github.Commit
	InitiatingUser github.User
	Root           terraform.Root
	Repo           github.Repo
	Tags           map[string]string
	Notifications  NotificationConfig
}
```